index.js

```diff
@@ -121,22 +122,33 @@ export default class VideoPlayer extends Component {
     this.onSeekRelease = this.onSeekRelease.bind(this);
     this.onSeek = this.onSeek.bind(this);
     this.onSeekByPosition = this.onSeekByPosition.bind(this);
+    this.activateStatistics = this.activateStatistics.bind(this);
+    this.deactivateStatistics = this.deactivateStatistics.bind(this);
   }
 
   componentDidMount() {
     if (this.props.autoplay) {
       this.hideControls();
+      this.props.stats && this.setState({ stats: true })
     }
   }
 
   componentWillUnmount() {
-    stats.deactivate()
+    this.deactivateStatistics()
     if (this.controlsTimeout) {
       clearTimeout(this.controlsTimeout);
       this.controlsTimeout = null;
     }
   }
 
+  deactivateStatistics () {
+    this.state.stats && this.deactivateStatistics()
+  }
+
+  collectStatistics () {
+    this.state.stats && this.state.isPlaying && stats.add(this.playerMode).activate()
```

Comment:
Is here really need check isPlaying state? In code this method called after when isPlaying state changed.

Reply:
There is one condition when it's called on duration changes
`
this.setState({ duration }, () => this.state.isPlaying && !this.wasPlayingBeforeSeek
      ? stats.add(this.playerMode).activate()
      : null
    )
`

But in the last commit I merged all checks in one function